lib/shoulda/matchers/active_model/validator.rb

```diff
@@ -0,0 +1,113 @@
+module Shoulda
+  module Matchers
+    module ActiveModel
+      # @private
+      class Validator
+        include Helpers
+
+        attr_writer :attribute, :context, :record
+
+        def initialize
+          reset
+        end
+
+        def reset
+          @messages = nil
+        end
+
+        def strict=(strict)
+          @strict = strict
+
+          if strict
+            extend StrictValidator
+          end
+        end
+
+        def capture_range_error(exception)
+          @captured_range_error = exception
+          extend ValidatorWithCapturedRangeError
+        end
+
+        def allow_description(allowed_values)
+          "allow #{attribute} to be set to #{allowed_values}"
+        end
+
+        def expected_message_from(attribute_message)
+          attribute_message
+        end
+
+        def messages
+          @messages ||= collect_messages
+        end
+
+        def formatted_messages
+          messages
+        end
+
+        def has_messages?
+          messages.any?
+        end
+
+        def messages_description
+          if has_messages?
+            " errors:\n#{pretty_error_messages(record)}"
+          else
+            ' no errors'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.